File: app/src/main/java/com/stathis/elmepaunivapp/ui/students/StudentsActivity.java
package com.stathis.elmepaunivapp.ui.students;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.stathis.elmepaunivapp.abstraction.AbstractActivity;
import com.stathis.elmepaunivapp.listeners.activity_listeners.StudentsActivityListener;
import com.stathis.elmepaunivapp.ui.dashboard.DashboardActivityKt;
import com.stathis.elmepaunivapp.ui.department.DepartmentActivity;
import com.stathis.elmepaunivapp.ui.professors.ProfessorsActivity;
import com.stathis.elmepaunivapp.R;
import com.stathis.elmepaunivapp.ui.students.model.Schedule;
import com.stathis.elmepaunivapp.ui.students.model.UsefulLinks;
import com.stathis.elmepaunivapp.ui.syllabus.SyllabusActivity;
import com.stathis.elmepaunivapp.ui.webview.WebviewActivity;

public class StudentsActivity extends AbstractActivity implements StudentsActivityListener {

    private RecyclerView recyclerView;
    private BottomNavigationView bottomNavigationView;
    private StudentsViewModel viewModel;

    public StudentsActivity() {
        super(R.layout.activity_students);
    }

    @Override
    public void initial() {
        viewModel = new ViewModelProvider(this).get(StudentsViewModel.class);

        recyclerView = findViewById(R.id.students_activity_recycler);
        bottomNavigationView = findViewById(R.id.bottom_nav);
    }

    @Override
    public void running() {
        recyclerView.setAdapter(viewModel.adapter);
        viewModel.createList(this);

        bottomNavigationView.setSelectedItemId(R.id.nav_students);
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                switch (item.getItemId()) {
                    case R.id.nav_home:
                        startActivity(new Intent(StudentsActivity.this, DashboardActivityKt.class));
                        break;
                    case R.id.nav_students:
                        return true;
                    case R.id.nav_uni:
                        startActivity(new Intent(StudentsActivity.this, DepartmentActivity.class));
                        break;
                    case R.id.nav_search:
                        startActivity(new Intent(StudentsActivity.this, ProfessorsActivity.class));
                        break;
                }
                return false;
            }
        });
    }

    @Override
    public void stopped() {

    }

    @Override
    public void openStudentsSchedule(Schedule schedule) {
        startActivity(new Intent(this, WebviewActivity.class).putExtra("URL", "https://mst.hmu.gr/proptyxiako/orologio-programma-mathimaton/"));
    }

    @Override
    public void itemClicked(UsefulLinks usefulLinks) {
        startActivity(new Intent(this, WebviewActivity.class).putExtra("URL", usefulLinks.getUrl()));
    }

    @Override
    public void goToSyllabus(UsefulLinks usefulLinks) {
        switch(usefulLinks.getName()){
            case "Επιστήμη των Δεδομένων & Τεχνολογίες Πληροφορικής" : {
                startActivity(new Intent(this, SyllabusActivity.class).putExtra("userTabChoice", 0));
                break;
            }
            case "Διοίκηση Επιχειρήσεων & Οργανισμών" : {
                startActivity(new Intent(this, SyllabusActivity.class).putExtra("userTabChoice",1));
                break;
            }
            case "Ψηφιακό Μάρκετινγκ και Επικοινωνία" : {
                startActivity(new Intent(this, SyllabusActivity.class).putExtra("userTabChoice",2));
                break;
            }
        }
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/students/StudentsActivity.kt
package com.stathis.elmepaunivapp.ui.students

import android.content.Intent
import androidx.lifecycle.ViewModelProvider
import com.stathis.elmepaunivapp.R
import com.stathis.elmepaunivapp.abstraction.ElmepaActivity
import com.stathis.elmepaunivapp.callbacks.StudentsScreenCallback
import com.stathis.elmepaunivapp.ui.dashboard.DashboardActivity
import com.stathis.elmepaunivapp.ui.department.DepartmentActivity
import com.stathis.elmepaunivapp.ui.professors.ProfessorActivity
import com.stathis.elmepaunivapp.ui.students.model.UsefulLinks
import com.stathis.elmepaunivapp.ui.syllabus.SyllabusActivity
import com.stathis.elmepaunivapp.ui.webview.WebviewActivity
import kotlinx.android.synthetic.main.activity_students.*
import kotlinx.android.synthetic.main.activity_webview.*
import kotlinx.android.synthetic.main.activity_webview.bottom_nav

class StudentsActivity : ElmepaActivity(R.layout.activity_students) {

    private lateinit var viewModel : StudentsViewModel

    override fun init() {
        viewModel = ViewModelProvider(this).get(StudentsViewModel::class.java)
    }

    override fun startOps() {
        viewModel.createLists()
        students_activity_recycler.adapter = viewModel.adapter

        viewModel.bindCallbacks(object : StudentsScreenCallback {
            override fun openSchedule() {
                startActivity(Intent(this@StudentsActivity, WebviewActivity::class.java)
                    .putExtra("URL", "https://mst.hmu.gr/proptyxiako/orologio-programma-mathimaton/"))
            }

            override fun openLink(usefulLinks: UsefulLinks) {
                startActivity(Intent(this@StudentsActivity, WebviewActivity::class.java)
                    .putExtra("URL", usefulLinks.url))
            }

            override fun openSyllabus(usefulLink: UsefulLinks) {
                goToSyllabus(usefulLink)
            }
        })

        setupBottomNavigation()
    }

    private fun goToSyllabus(data: UsefulLinks) {
        when(data.name){
            "Επιστήμη των Δεδομένων & Τεχνολογίες Πληροφορικής" -> {
                startActivity(Intent(this, SyllabusActivity::class.java)
                    .putExtra("userTabChoice", 0))
            }
            "Διοίκηση Επιχειρήσεων & Οργανισμών" -> {
                startActivity(Intent(this, SyllabusActivity::class.java)
                    .putExtra("userTabChoice", 1))
            }
            "Ψηφιακό Μάρκετινγκ και Επικοινωνία" -> {
                startActivity(Intent(this, SyllabusActivity::class.java)
                    .putExtra("userTabChoice", 2))
            }
        }
    }

    private fun setupBottomNavigation() {
        bottom_nav.setOnNavigationItemSelectedListener {
            when (it.itemId) {
                R.id.nav_home -> {
                    startActivity(Intent(this, DashboardActivity::class.java))
                    true
                }
                R.id.nav_students -> true

                R.id.nav_uni -> {
                    startActivity(Intent(this, DepartmentActivity::class.java))
                    true
                }

                R.id.nav_search -> {
                    startActivity(Intent(this, ProfessorActivity::class.java))
                    true
                }

                else -> false
            }
        }
    }

    override fun stopOps() {}
}